gcovr/__main__.py

```diff
@@ -425,17 +430,19 @@ def print_reports(covdata, options, logger):
                 "Coveralls output skipped - "
                 "consider providing output file with `--coveralls=OUTPUT`.")))
 
+    generator_error_occurred = False
     reports_were_written = False
     default_output_used = False
     default_output = OutputOrDefault(None) if options.output is None else options.output
+
     for output_choices, generator, on_no_output in generators:
         output = OutputOrDefault.choose(output_choices, default=default_output)
         if output is not None and output is default_output:
             default_output_used = True
             if not output.is_dir:
                 default_output = None
         if output is not None:
-            generator(covdata, output.abspath, options)
+            generator_error_occurred = generator(covdata, output.abspath, options) or generator_error_occurred
```

Comment:
To see that the errors are or'ed.
```suggestion
            generator_error_occurred = generator_error_occurred or generator(covdata, output.abspath, options)
```

Reply:
I wouldn't suggest to do that because it's logically different. If I changed to `generator_error_occurred or generator()` than if error occurred generator won't be called.
```python
>>> def go():
...   print 'asdf'
...
>>> True or go()
True
>>> go() or True
asdf
True
```

Reply:
Good point. If equating on left side of a or is True, the right site isn't evaluated.
I think a if clause, setting the flag in the body is clearer.